### PR TITLE
[DOC] Tweaks for Array#cycle

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -6782,13 +6782,18 @@ rb_ary_cycle_size(VALUE self, VALUE args, VALUE eobj)
...
 /*
  *  call-seq:
- *    array.cycle {|element| ... } -> nil
- *    array.cycle(count) {|element| ... } -> nil
- *    array.cycle -> new_enumerator
- *    array.cycle(count) -> new_enumerator
- *
- *  When called with positive Integer argument +count+ and a block,
- *  calls the block with each element, then does so again,
+ *    cycle {|element| ... } -> nil
```
Comment (peterzhu2118) [inline array.c:6785]: I don't think we need this case since the case below encompasses this one.

File: array.c [hunk 2]
```diff
@@ -6782,13 +6782,18 @@ rb_ary_cycle_size(VALUE self, VALUE args, VALUE eobj)
...
+ *    cycle(count = nil) {|element| ... } -> nil
+ *    cycle -> new_enumerator
```
Comment (peterzhu2118) [inline array.c:6787]: Ditto

File: array.c (hunk 1 above)
Comment (BurdetteLamar) [reply array.c]: Removed.

File: array.c (hunk 2 above)
Comment (BurdetteLamar) [reply array.c]: Ditto.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**